pages/api/fetchData.js

```diff
@@ -1,22 +1,44 @@
-export default async function fetchData() {
-  try {
-    const host = "http://api.wisey.app";
-    const version = "api/v1";
-    const accessToken = await fetch(
-      `${host}/${version}/auth/anonymous?platform=subscriptions`
-    );
-    const accesData = await accessToken.json();
-    const token = accesData.token;
-    const res = await fetch(`${host}/${version}/core/preview-courses`, {
-      headers: {
-        Authorization: `Bearer ${token}`,
-      },
-    });
-    const data = await res.json();
+import fetchAccessToken from "./accessToken";
+class ApiClient {
+  static instance;
+  accessToken;
```

Comment:
You can use private fields to make accessToken truly private

[https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields)

Reply:
Done)